src/soupBowl.ts

```diff
@@ -82,6 +84,7 @@ class SoupBowl {
             }).catch(error => {
                 Logger.error(error, this);
             });
+            reject(new Error('Request failed'));
```

Comment:
Will that not always reject the request right away? It looks like this reject will be called before resolve can be executed.

Reply:
Meh, you're right. :facepalm: